Rewrite discussion conclusion (#68)
greenelab/text_mined_hetnet_manuscript: content/05.discussion_and_conclusion.md

An exception to this trend is using label functions designed from conceptually related edge types (using GiG label functions to predict CbG sentences and vice versa).
Furthermore, broad edge types such as DaG did not follow this trend as we found this edge to be agnostic to any tested label function source.
One possibility for this observation is that the "associates" relationship is a general concept that may include other concepts such as Disease (up/down) regulating a Gene (examples highlighted in our [annotated sentences](https://github.com/greenelab/text_mined_hetnet_manuscript/tree/master/supplementary_materials/annotated_sentences)).

> These two results suggest that the structure of biomedical language contains similar text when describing straightforward relationship types and vice versa for broader relationship types.

Reviewer: ```suggestion
These two results suggest that the transferability of label functions is likely to relate to the nature of the edge type in question, so determining how many label functions will be required to scale across multiple relationship types will depend on how conceptually similar those types are..
```

> Similarly, we hypothesize that this textual property will hold when extending this work to incorporate other relationship types such as  Disease presents Symptom or Compound causes Side Effects.

Reviewer: ```suggestion
```
See if I got the take home into one (long) sentence above.

The discriminator model did not have an apparent positive or negative effect on performance; however, we noticed that performance heavily depended on the annotations provided by the generative model.
This pattern suggests a focus on label function construction and generative model training may be key steps to focus on in future work.
Although we found that label functions cannot be re-used across all edge types with the standard task framing, strategies like multitask [@doi:10.1145/3209889.3209898] or transfer learning [@doi:10.1186/s40537-016-0043-6] may make multi-label-function efforts more successful.

## Conclusions

> Regarding knowledge graph construction, the amount of work required will scale linearly based on the number of various edge types.

Reviewer: ```suggestion
```

> When using this framework one should expect to have at least 25-30 different label functions per relationship type.

Reviewer: ```suggestion
We found that performance often increased through the tested range of 25-30 different label functions per relationship type.
Our finding of limited value for reuse across most edge type pairs suggests that the amount of work required to construct graphs will scale linearly based on the number of edge types.
```

> During the label function design process, one should be cautious about the language used to describe relationship types.

Reviewer: ```suggestion
We did not investigate whether certain individual label functions, as opposed to the full set of label functions for an edge type, were particularly reusable.
It remains possible that some functions are generic and could be used as the base through supplementation with additional, type-specific, functions.
```

> For example, a broad biomedical relationship may use a plethora of textual keywords and phrases to describe the relationship.

Reviewer: ```suggestion
```

> Overall, we expect similar frameworks to be incorporated in future endeavors.

Reviewer: ```suggestion
Literature continues to grow at a rate likely to surpass what is feasible by human curation.
Further work is needed to understand how to automatically extract large-scale knowledge graphs from the wealth of biomedical text.
```
